File: 2st/Yeonwoo/주사위 게임 3.py
def solution(a, b, c, d):
    answer = 0
    arr = [a,b,c,d]
    arr.sort()
    if arr[0]==arr[3]:
        answer = 1111*arr[0]
    elif arr[0]==arr[2] or arr[1]==arr[3]:
        if arr[0]==arr[2]:
            answer = (10*arr[0]+arr[3])**2
        elif arr[1]==arr[3]:
            answer = (10*arr[1]+arr[0])**2
    elif arr[0]==arr[1] and arr[2]==arr[3]:
        answer = (arr[0]+arr[2])*abs(arr[0]-arr[2])
    elif arr[0]==arr[1]:
        answer = arr[2]*arr[3]
    elif arr[1]==arr[2]:
        answer = arr[0]*arr[3]
    elif arr[2]==arr[3]:
        answer = arr[0]*arr[1]
    else:
        answer = min(arr)
    return answer

File: 4st/Eunseong/주사위 게임 3.java
class Solution {
    public int solution(int a, int b, int c, int d) {
        int answer = 0;
        
        if (a == b && b == c && c == d) answer = 1111 * a;
        
        else if (a == b && b == c) answer = (int)Math.pow((10 * a + d), 2);
        else if (a == b && b == d) answer = (int)Math.pow((10 * a + c), 2);
        else if (a == c && c == d) answer = (int)Math.pow((10 * a + b), 2);
        else if (b == c && c == d) answer = (int)Math.pow((10 * b + a), 2);
        
        else if (a == b && c == d) answer = (a + c) * Math.abs(a - c);
        else if (a == c && b == d) answer = (a + b) * Math.abs(a - b);
        else if (b == c && a == d) answer = (b + a) * Math.abs(b - a);
        else if (a == d && b == c) answer = (a + b) * Math.abs(a - b);
        
        else if (a == b && c != d) answer = c * d;
        else if (a == c && b != d) answer = b * d;
        else if (a == d && b != c) answer = b * c;
        else if (b == c && a != d) answer = a * d;
        else if (b == d && a != c) answer = a * c;
        else if (c == d && a != b) answer = a * b;
        
        else if (a != b && b != c && c != d) answer = Math.min(b, Math.min(a, Math.min(c, d)));
        
        return answer;
    }
}